prov/rxm/src/rxm_ep.c

```diff
@@ -2248,8 +2231,22 @@ static void rxm_ep_settings_init(struct rxm_ep *rxm_ep)
 	rxm_ep->msg_mr_local = ofi_mr_local(rxm_ep->msg_info);
 	rxm_ep->rxm_mr_local = ofi_mr_local(rxm_ep->rxm_info);
 
-	rxm_ep_set_inject_buf_limit(rxm_ep, rxm_ep->rxm_info->tx_attr->inject_size);
-	assert(rxm_ep_get_inject_buf_limit(rxm_ep) <= rxm_ep->msg_info->tx_attr->inject_size);
+	assert(rxm_ep->rxm_info->tx_attr->inject_size != SIZE_MAX);
+	rxm_ep_set_inject_buf_limit(rxm_ep, (rxm_ep->rxm_info->tx_attr->inject_size ?
+					     (rxm_ep->rxm_info->tx_attr->inject_size + 1) : 0));
+	assert((rxm_ep_get_inject_buf_limit(rxm_ep) <= rxm_ep->msg_info->tx_attr->inject_size) &&
+	       /* This verifies that the following condition is met:
+		* if (`inject_size` of the MSG provider == RxM header size)
+		*        RxM inject size = 0
+		* else
+		*        RxM inject size = (`inject_size` of the MSG provider -
+		*                           RxM header size) + 1
+		*
+		* This is needed to compare user's buffer length with RxM inject limit
+		* using "<" instead of "<=" to ensure that RxM doesn't use `fi_inject()`
+		* in case of (`inject_size` of the MSG provider == 0).
+		*/
+	       (rxm_ep_get_inject_buf_limit(rxm_ep) != 1));
```

Comment:
I'm not sure if I understand this patch. Is the issue, application using fi_inject when RxM inject size is 0?